keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -333,3 +333,202 @@ features makes their future removal much harder.
 ### Release klog 3.0 with removed features
 Removal of those features cannot be done without whole k8s community instead of
 just k8s core components
+
+### Upgrade / Downgrade Strategy
+
+Proposed user impacting changes are as follows:
+* Deprecate and remove subset of klog flags
+* Split log output (previously only stderr) into stdout/stderr based on type (info/error)
+
+In case of the first change, we will be following K8s deprecation policy. There
+will be 3 releases between informing users about deprecation and full removal.
+During deprecation period there will not be any changes in behavior for clusters
+using deprecated features, however after removal there will not be a way to
+restore previous behavior. 3 releases should be enough heads up for users to
+make necessary changes to avoid breakage. 
+
+In case of changing log output, we will introduce a temporary flag 
+`--logtostdout` that will allow users to flip back to previous behavior. 
```

Comment:
Previous behaviour was logging to stderr, not stdout. Also, do we really want to add a new flag only just to deprecate it in the shortest possible transition period? This seems unusually disruptive.

Edit: ah, this is discussed below where it states the flag will introduce what will become the new default.

I don't think there's sufficient differentiation in k8s logs between errors and info to merit this split. There are still a number of log lines that are marked as errors but recoverable/no need to mention them to user iirc... I'm just not seeing the benefit for the corresponding ops churn. Maybe you can elaborate?

Reply:
Discussed motivation in comment above.